themes/default/templates/macros.mtt

```diff
@@ -101,9 +101,9 @@
 		$$printTypeParams(::field.params::) (
 		::foreach arg args[0]::
 			<span style="white-space:nowrap">
-			::if arg.opt && arg.value == null::?::end::
-			::arg.name:::$$printLinkedType(::arg.t::)
-			::if arg.value != null && arg.value != "null":: = ::arg.value::::end::
+				::if arg.opt && (arg.value == null || arg.value == "null")::?::end::
```

Comment:
This is the naive implementation without any checks:

```
::if arg.opt::?::end::
::arg.name:::$$printLinkedType(::arg.t::)
 = ::arg.value::
```

The result from that looks like this:

![](http://i.imgur.com/zP3IgIn.png)

(Wrong / confusing for `a` and `c`).

I guess the value being null is a special case because the result of that is pretty redundant - both a `?` and `= null`, so `= null` is omitted.

`?b = "foo"` is printed as `b:String = "foo"` right now (https://github.com/dpeek/dox/issues/68#issuecomment-42023843).